python/pywatchman/__init__.py

```diff
@@ -690,10 +690,17 @@ def _resolvesockname(self):
 
         cmd = ['watchman', '--output-encoding=bser', 'get-sockname']
         try:
+            # if invoked via an application with a graphical user interface,
+            # this call will cause a brief command window pop-up.
+            # Using the flag STARTF_USESHOWWINDOW to avoid this behavior.
+            startupinfo = subprocess.STARTUPINFO()
+            startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
+            
             p = subprocess.Popen(cmd,
                                  stdout=subprocess.PIPE,
                                  stderr=subprocess.PIPE,
-                                 close_fds=os.name != 'nt')
+                                 close_fds=os.name != 'nt',
+-                                startupinfo=startupinfo)
```

Comment:
Thanks, I am already fixing it.

On Mon, Jul 18, 2016 at 4:16 PM, Wez Furlong <notifications@github.com>
wrote:

> In python/pywatchman/__init__.py
> <https://github.com/facebook/watchman/pull/304#discussion_r71220779>:
>
> >              p = subprocess.Popen(cmd,
> >                                   stdout=subprocess.PIPE,
> >                                   stderr=subprocess.PIPE,
> > -                                 close_fds=os.name != 'nt')
> > +                                 close_fds=os.name != 'nt',
> > +-                                startupinfo=startupinfo)
>
> the stray - here is causing some problems
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/watchman/pull/304/files/d68093b2f7b758622802829ccfc48d5181c80372#r71220779>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADp6JPzCbXZqAyI68WnfH9QhSLfFvTkzks5qW98ygaJpZM4JPFkV>
> .
>